File: Python/Punto_10.py
# Se importa el módulo numpy y se le asigna el alias 'np'
import numpy as np

# Se crea una matriz de 4 filas y 5 columnas con valores enteros aleatorios entre -100 y 99
matriz = np.random.randint(-100, 100, size=(4, 5))

# Se imprime la matriz generada
print("Matriz:")
print(matriz)

# Se solicita al usuario que ingrese un número entero para buscarlo en la matriz
valor = int(input("Ingrese un valor: "))

"""Se utiliza la función np.where para obtener las posiciones (índices) donde el valor ingresado
   aparece en la matriz
   Esta función retorna dos arreglos: uno con las filas y otro con las columnas donde se encontró
   el valor"""
posicion = np.where(matriz == valor)

# Se verifica si el valor fue encontrado (es decir, si el arreglo de filas no está vacío)
if posicion[0].size > 0:
    # Si se encontró, se muestra la primera posición donde aparece el valor
    print(f"El valor {valor} se encuentra en la posición ({posicion[0][0]}, {posicion[1][0]})")
else:
    # Si no se encontró, se muestra un mensaje indicándolo
    print(f"El valor {valor} no se encuentra en la matriz")

File: Java/Punto10.java
package Java;

import java.util.Arrays;
import java.util.Random;
import java.util.Scanner;
import java.util.stream.IntStream;

public class Punto10 {
    public static void main(String[] args) {
        int filas = 4, columnas = 5;
        Random random = new Random();
        Scanner scanner = new Scanner(System.in); // La entrada del usuario
        int[][] matriz = IntStream.range(0, filas)
                .mapToObj(i -> IntStream.range(0, columnas)
                        .map(j -> random.nextInt(100) + 1)
                        .toArray())
                .toArray(int[][]::new);

        Arrays.stream(matriz)
                .map(Arrays::toString)
                .forEach(System.out::println);

        System.out.println("Ingrese un número para buscar en la matriz: ");
        int numeroBuscado = scanner.nextInt();
        IntStream.range(0, filas * columnas)
                .filter(i -> matriz[i / columnas][i % columnas] == numeroBuscado)
                /*Indica que se recorre índice por índice, donde corresponda el número buscado al que esta en ese espacio
                * Funciona, por índices.*/

                .mapToObj(i -> "Posición: [" + (i / columnas) + "][" + (i % columnas) + "]")
                .forEach(System.out::println);
    }
}